modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/view_discard.jsp

```diff
@@ -10,7 +10,9 @@
 <%
 ViewRelatedEntriesDisplayContext viewRelatedEntriesDisplayContext = (ViewRelatedEntriesDisplayContext)request.getAttribute(CTWebKeys.VIEW_RELATED_ENTRIES_DISPLAY_CONTEXT);
 
-portletDisplay.setURLBack(viewRelatedEntriesDisplayContext.getRedirectURL());
+String backURL = ParamUtil.getString(request, "backURL");
```

Comment:
Use redirectURL as the default for backURL in case the backURL is not specifically set in the view discard url. And use the backURL for the cancel button.